crates/ruff_cli/tests/integration_test.rs

```diff
@@ -861,11 +789,223 @@ fn check_input_from_argfile() -> Result<()> {
     });
 
     Ok(())
 }
+
+#[test]
+fn displays_fix_applicability_levels() {
+    // `--fix` should only apply automatic fixes, but should tell the user about `--fix-suggested` if
+    // there are remaining fixes.
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args([
+            "-",
+            "--output-format=text",
+            "--isolated",
+            "--select",
+            "F601,UP034",
+            "--no-cache",
+        ])
+        .pass_stdin("x = {'a': 1, 'a': 1}\nprint(('foo'))\n"),
+        @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    -:1:14: F601 [**] Dictionary key literal `'a'` repeated
+    -:2:7: UP034 [*] Avoid extraneous parentheses
+    Found 2 errors.
+    [*] 1 fixable with the --fix option.
+    [**] 1 potentially fixable with the --fix-suggested option.
+
+    ----- stderr -----
+    "###);
+}
+
+#[test]
+fn displays_remaining_suggested_fixes() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["-", "--output-format", "text", "--no-cache", "--isolated", "--select", "F601"])
+        .pass_stdin("x = {'a': 1, 'a': 1}\n"),
+        @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    -:1:14: F601 [**] Dictionary key literal `'a'` repeated
+    Found 1 error.
+    [**] 1 potentially fixable with the --fix-suggested option.
+
+    ----- stderr -----
+    "###);
+}
+
+#[test]
+fn fix_applies_automatic_fixes_only() {
+    // `--fix` should only apply automatic fixes. Since we're runnnig in `stdin` mode, output shouldn't
+    // be printed.
+    assert_cmd_snapshot!(
+        Command::new(get_cargo_bin(BIN_NAME))
+            .args([
+                "-",
+                "--output-format",
+                "text",
+                "--isolated","--no-cache", 
+                "--select",
+                "F601,UP034",
+                "--fix",
+            ])
+            .pass_stdin("x = {'a': 1, 'a': 1}\nprint(('foo'))\n"),
+            @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    x = {'a': 1, 'a': 1}
+    print('foo')
+
+    ----- stderr -----
+    "###);
+}
+
+#[test]
+fn fix_applies_automatic_and_suggested_fixes_with_fix_suggested() {
+    assert_cmd_snapshot!(
+        Command::new(get_cargo_bin(BIN_NAME))
+            .args([
+                "-",
+                "--output-format",
+                "text",
+                "--isolated","--no-cache", 
+                "--select",
+                "F601,UP034",
+                "--fix",
+                "--fix-suggested",
+            ])
+            .pass_stdin("x = {'a': 1, 'a': 1}\nprint(('foo'))\n"),
+            @r###"
+    success: false
+    exit_code: 1
+    ----- stdout -----
+    -:1:14: F601 [**] Dictionary key literal `'a'` repeated
+    -:2:7: UP034 [*] Avoid extraneous parentheses
+    Found 2 errors.
+    [*] 1 fixable with the --fix option.
+    [**] 1 potentially fixable with the --fix-suggested option.
```

Comment:
That's confusing, the user just passed `--fix-suggested` and now we're telling them to use `--fix-suggested`, also shouldn't we get the fixed source back?

Reply:
Hmm yeah, what's going on here? :)